frame/lending/src/lib.rs

```diff
@@ -688,119 +859,55 @@ pub mod pallet {
-			<Self as Lending>::get_borrow_limit(market_id, account)
-		}
+	/////////////////////////////////////////////////////////////////////////
+	// Public helper functions
+	/////////////////////////////////////////////////////////////////////////
 
-		pub fn borrow_internal(
-			market_id: &<Self as Lending>::MarketId,
-			debt_owner: &<Self as DeFiEngine>::AccountId,
-			amount_to_borrow: <Self as DeFiEngine>::Balance,
-		) -> Result<(), DispatchError> {
-			<Self as Lending>::borrow(market_id, debt_owner, amount_to_borrow)
+	impl<T: Config> Pallet<T> {
+		/// Returns the initial pool size for a a market with `borrow_asset`. Calculated with
+		/// [`Config::OracleMarketCreationStake`].
+		pub fn calculate_initial_pool_size(
+			borrow_asset: <T::Oracle as composable_traits::oracle::Oracle>::AssetId,
+		) -> Result<<T as composable_traits::defi::DeFiComposableConfig>::Balance, DispatchError> {
+			T::Oracle::get_price_inverse(borrow_asset, T::OracleMarketCreationStake::get())
 		}
 
-		pub fn borrow_balance_current(
+		/// Creates a new [`BorrowerData`] for the given market and account. See [`BorrowerData`]
+		/// for more information.
+		pub fn create_borrower_data(
 			market_id: &<Self as Lending>::MarketId,
 			account: &<Self as DeFiEngine>::AccountId,
-		) -> Result<Option<BorrowAmountOf<Self>>, DispatchError> {
-			<Self as Lending>::borrow_balance_current(market_id, account)
-		}
-
-		pub fn total_borrows(
-			market_id: &<Self as Lending>::MarketId,
-		) -> Result<<Self as DeFiEngine>::Balance, DispatchError> {
-			<Self as Lending>::total_borrows(market_id)
-		}
+		) -> Result<BorrowerData, DispatchError> {
+			let market = Self::get_market(market_id)?;
 
-		pub fn total_cash(
-			market_id: &<Self as Lending>::MarketId,
-		) -> Result<<Self as DeFiEngine>::Balance, DispatchError> {
-			<Self as Lending>::total_cash(market_id)
-		}
+			let collateral_balance_value = Self::get_price(
+				market.collateral_asset,
+				Self::collateral_of_account(market_id, account)?,
+			)?;
 
-		pub fn total_interest(
-			market_id: &<Self as Lending>::MarketId,
-		) -> Result<<Self as DeFiEngine>::Balance, DispatchError> {
-			<Self as Lending>::total_interest(market_id)
-		}
+			let account_total_debt_with_interest =
+				Self::total_debt_with_interest(market_id, account)?.unwrap_or_zero();
+			let borrow_balance_value = Self::get_price(
+				T::Vault::asset_id(&market.borrow_asset_vault)?,
+				account_total_debt_with_interest,
+			)?;
 
-		pub fn repay_borrow_internal(
-			market_id: &<Self as Lending>::MarketId,
-			from: &<Self as DeFiEngine>::AccountId,
-			beneficiary: &<Self as DeFiEngine>::AccountId,
-			repay_amount: Option<BorrowAmountOf<Self>>,
-		) -> Result<(), DispatchError> {
-			<Self as Lending>::repay_borrow(market_id, from, beneficiary, repay_amount)
-		}
+			let borrower =
+				BorrowerData::new(
+					collateral_balance_value,
+					borrow_balance_value,
+					market
+						.collateral_factor
```

Comment:
you should not check it here. it should be checked only during creation/update, but not in operations. imagine MoreThan1(number). and that type ecodes property. so it will hold it always after creation. SO, all other places should not check as type enforced it when it was created